### DISCUSSION TITLE
Set labels on merge queue PRs based on the labels of the original PR(s)

### DISCUSSION THREAD

**Discussion by nixpanic** [Mergifyio/mergify#5088] [Ideas]
It would be very useful if labels for a newly created **merge queue** PR get inherited from the original PR(s). We use labels to mark some tests as (not) relevant, and we now manually need to set those labels on the **merge queue** PRs again. For certain tests, our CI requires ~2 hours, but on occasion those tests are not required and can be skipped (the CI job checks for the label and just skips testing when the label is set).

**Comment by jd** [comment]
Hi @nixpanic 

You can use this GHA https://github.com/Mergifyio/gha-mergify-merge-queue-labels-copier to solve this use case. :)

**Reply by nixpanic** [reply]
Thanks for the pointer! I'll try it out.

**Reply by nixpanic** [reply]
We added a GitHub Workflow, but it failed to copy labels for ceph/ceph-csi#3812.
There are no labels configured in the `with.labels:` section, in the understanding that all labels would be copied.

Any idea what we're doing wrong? This is the run that should have copied the `rebase` and `ci/skip/e2e` labels:  https://github.com/ceph/ceph-csi/actions/runs/4949868278/jobs/8852661810?pr=3812

**Reply by DouglasBlackwood** [reply]
I think you have to do something like this:
```yaml
  - name: Copying labels
    uses: Mergifyio/gha-mergify-merge-queue-labels-copier@main
    with:
      labels:
```
`labels` has to be an empty string.

**Reply by nixpanic** [reply]
We noticed that `gha-mergify-merge-queue-labels-copier` had some issues, and PRs were sent to address them. With the current version of the action it all works :partying_face:
